packages/cdktf-cli/lib/get/generator/provider-generator.ts

```diff
@@ -77,32 +78,41 @@ export class TerraformProviderGenerator {
       throw Errors.Internal("get", `can't handle ${fqpn}`, { fqpn });
     }
 
-    const files: string[] = [];
+    const resourceModels: ResourceModel[] = [];
     for (const [type, resource] of Object.entries(
       provider.resource_schemas || []
     )) {
-      files.push(
-        this.emitResourceFile(
-          this.resourceParser.parse(name, type, resource, "resource")
-        )
+      resourceModels.push(
+        this.resourceParser.parse(name, type, resource, "resource")
       );
     }
-
     for (const [type, resource] of Object.entries(
       provider.data_source_schemas || []
     )) {
-      files.push(
-        this.emitResourceFile(
-          this.resourceParser.parse(
-            name,
-            `data_${type}`,
-            resource,
-            "data_source"
-          )
-        )
+      resourceModels.push(
+        this.resourceParser.parse(name, `data_${type}`, resource, "data_source")
       );
     }
 
+    type NamespaceName = string;
+    const namespacedResources: Record<NamespaceName, ResourceModel[]> = {};
+    const files: string[] = [];
+    resourceModels.forEach((resourceModel) => {
+      if (resourceModel.namespace) {
+        const namespace = resourceModel.namespace.name;
+        if (!namespacedResources[namespace]) {
+          namespacedResources[namespace] = [];
+        }
+        namespacedResources[namespace].push(resourceModel);
+      } else {
+        files.push(this.emitResourceFile(resourceModel));
+      }
+    });
+
+    for (const [namespace, resources] of Object.entries(namespacedResources)) {
+      files.push(this.emitNamespacedResourceFile(namespace, resources));
+    }
+
```

Comment:
That's generating a single file for the namespace?

Reply:
Yeah as far as I could tell you can not reopen a namespace, it has to be in one block (I had multiple files with the same namespace before and the last one won)

Reply:
Cool 👍